### PR TITLE
Configure pnpm, eslint, prettier and Github action

### PR DESCRIPTION
**Pull request by jakeanq**
I've switched to `pnpm` because it's much faster than bare `npm` and a bit more modern.

I've also configured ESLint (JS/TS linting) and prettier (auto-formatting for just about everything except C++) and set them both up to run automatically on push in a Github action.

We'll need to add an additional config to ESLint for React/Svelte/etc once we make a decision there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
+
 - Node.js
+  - Install `nvm` by following the instructions here: https://github.com/nvm-sh/nvm#installing-and-updating
+- pnpm
+  - Install using `corepack`:
 
-## Development
-The **Frontend** and **CEF App** projects are set up to work from the root reporitory directory, so no directory changes are required to build/run both.
+```sh
+corepack enable
```
Comment (RVeips) [inline README.md]: `corepack enable` does not work if the user does not have permissions to write to /bin
Possible fixes:
1. `corepack enable --install-directory ~/bin` (`mkdir ~/bin` would be required for the command to work if ~/bin does not exist)
2. `sudo corepack enable`

**Review by RVeips — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jakeanq) [reply README.md]: I've updated to include the non-sudo version in the readme, as well updated the `nvm` notes slightly to hopefully make them a bit clearer - using `nvm` also avoids the above issue.

File: README.md [hunk 2]
```diff
@@ -4,18 +4,25 @@ This repository contains the default Excessive Motion UI frontend and a minimal
...
+- pnpm
+  - Install using ```corepack```:
+```sh
+corepack enable
+corepack prepare pnpm@latest
+```
+
+### Development
+Before starting, run ```pnpm install``` to download all of the development dependencies.
+- Run ```pnpm start``` to start a live development server at localhost:8080
```
Comment (RVeips) [inline README.md]: Small detail, but the live server is not always started on :8080 (if :8080 is already in use, the server will be started on the next available port)

**Comment by RVeips** [comment]
@jakeanq Small detail about the README - webpack live server is not always started on :8080 (if :8080 is already in use, the server will be started on the next available port)

**Comment by jakeanq** [comment]
@RVeips updated - I've removed the reference entirely since I figure it's pretty obvious with the web browser auto-opening.

**Pull request merged**